src/api/Controllers/ExportController.cs

```diff
@@ -56,7 +57,107 @@ public async Task<ActionResult> ExportJsonAsync([FromQuery][MaxLength(MaxPageSiz
-    /// Exports the details of up to <see cref="MaxPageSize"></see> boreholes as a CSV file. Filters the boreholes based on the provided list of ids.
+    /// Asynchronously gets all <see cref="Borehole"/> records filtered by ids. Additional data is included in the response.
+    /// </summary>
+    /// <param name="ids">The required list of borehole ids to filter by.</param>
+    [HttpGet("geopackage")]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    public async Task<ActionResult> ExportGeoPackageAsync([FromQuery][MaxLength(MaxPageSize)] IEnumerable<int> ids)
+    {
+        if (ids == null || !ids.Any()) return BadRequest("The list of IDs must not be empty.");
+
+        var boreholes = await context.Boreholes.GetAllWithIncludes().AsNoTracking().Where(borehole => ids.Contains(borehole.Id)).ToListAsync().ConfigureAwait(false);
+        var boreholeIds = boreholes.Select(b => b.Id).ToList();
+
+        var boreholeGeometries = await GetBoreholeGeometries(boreholeIds).ConfigureAwait(false);
+
+        List<object> features = new();
+        foreach (var borehole in boreholes)
+        {
+            borehole.SetTvdValues(boreholeGeometries);
+
+            var feature = new
+            {
+                type = "Feature",
+                geometry = new
+                {
+                    type = "Point",
+                    crs = new
+                    {
+                        type = "name",
+                        properties = new
+                        {
+                            name = "EPSG:2056",
+                        },
+                    },
+                    coordinates = new[]
+                    {
+                        borehole.LocationX,
+                        borehole.LocationY,
+                    },
+                },
+                properties = borehole,
+            };
+            features.Add(feature);
+        }
+
+        var geojson = new
+        {
+            type = "FeatureCollection",
+            crs = new
+            {
+                type = "name",
+                properties = new
+                {
+                    name = "EPSG:2056",
+                },
+            },
+            features,
+        };
+
+        try
+        {
+            var tempGeoJsonFilePath = Path.Combine(Path.GetTempPath(), $"temp_geojson_{Guid.NewGuid()}.geojson");
+            await System.IO.File.WriteAllTextAsync(tempGeoJsonFilePath, JsonSerializer.Serialize(geojson, jsonExportOptions)).ConfigureAwait(false);
+
+            GdalBase.ConfigureAll();
+            Ogr.RegisterAll();
+            Ogr.UseExceptions();
+
+            // Create a temporary file for GeoPackage
+            var tempGpkgFilePath = Path.Combine(Path.GetTempPath(), $"temp_gpkg_{Guid.NewGuid()}.gpkg");
+            var openFileGdbDriver = Ogr.GetDriverByName("GPKG");
+
+            // Write GeoPackage to temporary file
+            using (var gpkgDataSource = openFileGdbDriver.CreateDataSource(tempGpkgFilePath, null))
```

Comment:
Q: kann man hier die neue using syntax verwenden?

Reply:
Könnte man. Dann muss man aber `gpkgDataSource.Dispose()` vor der Zeile
`var gpkgBytes = await System.IO.File.ReadAllBytesAsync(tempGpkgFilePath).ConfigureAwait(false);`
aufrufen. Da sonst `ReadAllBytesAsync(tempGpkgFilePath)` fehlschlägt weil er die Datei lesen will welche noch vom Prozess gpkgDataSource verwendet. 
Was findest du besser?

```suggestion

using var gpkgDataSource = openFileGdbDriver.CreateDataSource(tempGpkgFilePath, null);

using var geojsonDataSource = Ogr.Open(tempGeoJsonFilePath, 1);

if (geojsonDataSource == null)
{
    throw new InvalidOperationException("Could not open input GeoJSON datasource.");
}

gpkgDataSource.CopyLayer(geojsonDataSource.GetLayerByIndex(0), "boreholes", null);
gpkgDataSource.Dispose();

// Read GeoPackage into memory
var gpkgBytes = await System.IO.File.ReadAllBytesAsync(tempGpkgFilePath).ConfigureAwait(false);
```

Reply:
Dann bin ich dafür es so zu lassen, hatte das Problem auch schon so ähnlich 👍